### PR TITLE
PI-760: CF\API\AbstractPluginActions::login() returns error for bad e…

### PR DESCRIPTION
**Pull request by jwineman**
…mail/apiKey now.

### CODE REVIEW

File: src/API/AbstractPluginActions.php [hunk 1]
```diff
@@ -85,11 +85,19 @@ public function login()
         $isCreated = $this->dataStore->createUserDataStore($requestBody["apiKey"], $requestBody["email"], null, null);
 
         if (!$isCreated) {
-            $this->logger->error('Creating user data to store failed');
```
Comment (jwineman) [inline src/API/AbstractPluginActions.php:88]: `createAPIError()` calls `$this->logger->error()`

File: src/API/AbstractPluginActions.php [hunk 2]
```diff
@@ -85,11 +85,19 @@ public function login()
         $isCreated = $this->dataStore->createUserDataStore($requestBody["apiKey"], $requestBody["email"], null, null);
```
Comment (thellimist) [inline src/API/AbstractPluginActions.php:85]: It'd make more sense if we first check the apiKey then save it.

```
        //Make a test request to see if the API Key, email are valid
        $testRequest = new Request('GET', 'user/', array(), array());
        $testResponse = $this->clientAPI->callAPI($testRequest);
        if(!$this->clientAPI->responseOk($testResponse)) {
            // WE DON"T NEED THIS NOW
            //$this->dataStore->createUserDataStore(null, null, null, null);
            return $this->api->createAPIError("Email address or API key invalid.");
        }

        // Save credential
        $isCreated = $this->dataStore->createUserDataStore($requestBody["apiKey"], $requestBody["email"], null, null);
        if (!$isCreated) {
            return $this->api->createAPIError('Unable to save user credentials');
        }

        $response = $this->api->createAPISuccessResponse(array('email' => $requestBody["email"]));
        return $response;
```

Comment (jwineman) [reply src/API/AbstractPluginActions.php:85]: https://github.com/cloudflare/cloudflare-plugin-backend/blob/master/src/API/Client.php#L22-L23 sets the apikey/email based on dataStore values.  Have to create them some how or modify the api client.  

We could modify `CF\API\Request` to contain headers and modify `CF\API\Client` to only set them if they're not already set.

Comment (thellimist) [reply src/API/AbstractPluginActions.php:85]: Not worth the change. This looks fine.

**Pull request merged**